src/components/Header/Header.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef, Dispatch, SetStateAction } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  inputChange: string;
+  focusOnChange: boolean;
+  loadingTodoIds: number[];
+  onInputChange: Dispatch<SetStateAction<string>>;
+  toggleTodo: (todoId: number, updates: Partial<Todo>) => void;
+  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  toggleTodo,
+  inputChange,
+  handleSubmit,
+  onInputChange,
+  focusOnChange,
+  loadingTodoIds,
+}) => {
+  const todoInputField = useRef<HTMLInputElement>(null);
+
+  const allCompleted = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    if (todoInputField.current) {
+      todoInputField.current.focus();
+    }
+  }, [focusOnChange]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.